src/wavm.cpp

```diff
@@ -271,21 +279,28 @@ ExecutionResult WavmEngine::internalExecute(
   ensureCondition(mainFunction, ContractValidationFailure, "\"main\" not found");
 
   // this is how WAVM's try/catch for exceptions
-  Runtime::catchRuntimeExceptions(
-    [&] {
-      try {
-        vector<IR::Value> invokeArgs;
-        Runtime::invokeFunctionChecked(wavm_context, mainFunction, invokeArgs);
-      } catch (EndExecution const&) {
-        // This exception is ignored here because we consider it to be a success.
-        // It is only a clutch for POSIX style exit()
+  // TODO: this is terrible, need to clean up the handling of `interface`
+  try {
+    Runtime::catchRuntimeExceptions(
+      [&] {
+        try {
+          vector<IR::Value> invokeArgs;
+          Runtime::invokeFunctionChecked(wavm_context, mainFunction, invokeArgs);
+        } catch (EndExecution const&) {
+          // This exception is ignored here because we consider it to be a success.
+          // It is only a clutch for POSIX style exit()
+        }
+      },
+      [&](Runtime::Exception&& exception) {
+        // FIXME: decide if each of the exception fit into VMTrap/InternalError
+        ensureCondition(false, VMTrap, Runtime::describeException(exception));
       }
-    },
-    [&](Runtime::Exception&& exception) {
-      // FIXME: decide if each of the exception fit into VMTrap/InternalError
-      ensureCondition(false, VMTrap, Runtime::describeException(exception));
-    }
-  );
+    );
+  } catch (exception const&) {
+    // clean up
+    wavm_host_module::interface.pop();
```

Comment:
This is actually fixing the problem of exceptions not cleaning up between calls.